### PR TITLE
add return_errors feature to hist2array

### PR DESCRIPTION
**Pull request by douglasdavis**
This takes the sqrt(sumw2) recommendation from #302 but doesn't implement tests
I'm submitting another PR because I ended up deleting the repo connected with #302, it was my first time setting up Travis CI and I felt like starting from scratch was a lot cleaner and more desirable than keeping my mess of commits which came from playing around playing with the CI.

Definitely a small world seeing Mark in Australia!

### CODE REVIEW

**Comment by ngauvin** [comment]
Hi,
It would be really great to have that feature in the next release. I currently convert the ROOT histogram (content and low edges) using root_numpy and then convert the errors by hand.
I see that the merge request is pending since a long time... What prevents it ?
Thanks.

**Comment by eduardo-rodrigues** [comment]
Do you know more on this @kratsg? Indeed it seems this could be dealt with reasonably quickly.

File: root_numpy/_hist.py [hunk 1]
```diff
@@ -226,6 +232,21 @@ def hist2array(hist, include_overflow=False, copy=True, return_edges=False):
...
+            if hist_type == 'C':
+                raise TypeError("THxC cannot return Errors")
+            else:
+                # the sum of weights squared is stored in a ROOT
+                # TArrayD pointer. So we must use key 'D' from the
+                # DTYPE_ROOT2NUMPY dictionary.
+                errors = np.sqrt(np.ndarray(shape=shape, dtype=DTYPE_ROOT2NUMPY['D'],
+                                            buffer=hist.GetSumw2().GetArray()))
+        else:
+            raise TypeError("return_errors does not support THn, THnSparce")
```
Comment (eduardo-rodrigues) [inline root_numpy/_hist.py:247]: ``

Comment (eduardo-rodrigues) [inline root_numpy/_hist.py:247]:
```suggestion
            raise TypeError("return_errors does not support THn, THnSparse")
```

File: root_numpy/_hist.py [hunk 2]
```diff
@@ -226,6 +232,21 @@ def hist2array(hist, include_overflow=False, copy=True, return_edges=False):
...
+            if hist_type == 'C':
+                raise TypeError("THxC cannot return Errors")
+            else:
+                # the sum of weights squared is stored in a ROOT
+                # TArrayD pointer. So we must use key 'D' from the
+                # DTYPE_ROOT2NUMPY dictionary.
+                errors = np.sqrt(np.ndarray(shape=shape, dtype=DTYPE_ROOT2NUMPY['D'],
+                                            buffer=hist.GetSumw2().GetArray()))
+        else:
+            raise TypeError("return_errors does not support THn, THnSparse")
```
Comment (eduardo-rodrigues) [inline root_numpy/_hist.py:247]:
```suggestion
            raise TypeError("return_errors does not support THn and THnSparse")
```

**Comment by eduardo-rodrigues** [comment]
Just a trivial change to retrigger the CI and check things after 2 years, as this seems pretty ready to be merged.

**Comment by eduardo-rodrigues** [comment]
Hi @kratsg, though Travis is not exploitable at this moment I see nothing strange here and would be tempted to merge this for a potentially final release taking on board the 20-ish commits not in the latest version. Indeed we still get requests ... What do you think?

**Comment by kratsg** [comment]
Fine with me. We probably need to kill travis amongst the various scikit-hep repos at some point.

**Comment by eduardo-rodrigues** [comment]
OK, cool.

Yep, the bulk of the packages use GHA but not this one.

**Pull request merged**